tests/html/shim.html

```diff
@@ -16,102 +16,299 @@
+    }
+    ES5Element2.prototype = Object.create(HTMLElement.prototype);
+    ES5Element2.prototype.constructor = ES5Element2;
+
+    customElements.define('es5-element-2', ES5Element2);
+
+    const el = new ES5Element2();
+    assert.instanceOf(el, ES5Element2);
+    assert.instanceOf(el, HTMLElement);
+    assert.equal(el.tagName, 'ES5-ELEMENT-2');
+  });
+
+  test('works with parser created elements', () => {
+    function ES5Element3() {
+      return HTMLElement.apply(this);
+    }
+    ES5Element3.prototype = Object.create(HTMLElement.prototype);
+    ES5Element3.prototype.constructor = ES5Element3;
+
+    customElements.define('es5-element-3', ES5Element3);
+
+    const container = document.createElement('div');
+    container.innerHTML = '<es5-element-3></es5-element-3>';
+    const el = container.querySelector('es5-element-3');
+    assert.instanceOf(el, ES5Element3);
+    assert.instanceOf(el, HTMLElement);
+    assert.equal(el.tagName, 'ES5-ELEMENT-3');
+  });
+
+  test('reactions', () => {
+    function ES5Element4() {
+      return HTMLElement.apply(this);
+    }
+    ES5Element4.prototype = Object.create(HTMLElement.prototype);
+    ES5Element4.prototype.constructor = ES5Element4;
+    ES5Element4.observedAttributes = ['test'];
+    ES5Element4.prototype.connectedCallback = function() {
+      this.connectedCalled = true;
+    };
+    ES5Element4.prototype.disconnectedCallback = function() {
+      this.disconnectedCalled = true;
+    };
+    ES5Element4.prototype.attributeChangedCallback = function(name, oldValue, newValue) {
+      this.attributeChangedCalled = `${name}, ${oldValue}, ${newValue}`;
+    };
+    ES5Element4.prototype.adoptedCallback = function() {
+      this.adoptedCalled = true;
+    };
+
+    customElements.define('es5-element-4', ES5Element4);
+
+    const container = document.createElement('div');
+    document.body.appendChild(container);
+
+    const el = new ES5Element4();
+
+    container.appendChild(el);
+    assert(el.connectedCalled);
+
+    container.removeChild(el);
+    assert(el.disconnectedCalled);
+
+    el.setAttribute('test', 'good');
+    assert.equal(el.attributeChangedCalled, 'test, null, good');
+
+    const doc = document.implementation.createHTMLDocument();
+    doc.adoptNode(el);
+    assert(el.adoptedCalled);
+
+  });
+
+});
+
+suite('Native Shim with ES5 + Reflect API super call (can be expected from transpilers)', () => {
+
+  // Do nothing if the browser does not natively support custom elements.
+  if (!window.customElements) {
+    test.skip('The native shim does not apply to this browser.', () => {});
+    return;
+  }
+  // Do nothing if the browser does not support Reflect API.
+  if (typeof Reflect !== 'object' || typeof Reflect.construct !== 'function') {
+    test.skip('This test suite does not apply to this browser (Reflect API missing).', () => {});
+    return;
+  }
+
+  test('works with createElement()', () => {
+    function ES5ReflectReflect1() {
```

Comment:
This looks like it was meant to be `ES5ReflectElement1`. Same goes for `...2` and `...3`.

Reply:
Fixed.